rfd/0057-automatic-aws-server-discovery.md

```diff
@@ -0,0 +1,119 @@
+---
+authors: Alex McGrath (alex.mcgrath@goteleport.com
+state: draft
+---
+
+# RFD 57 - Automatic discovery and enrollment of AWS servers
+
+## What
+
+Proposes a way by which an auth service might automatically discover and
+register AWS EC2 instances.
+
+## Why
+
+Currently when adding a new AWS server, its required that Teleport be installed
+after the server has been provisioned which may be a slow process for
+organizations with large numbers of servers as it needs to be installed and then
+added to the teleport cluster
+
+With the changes described in this document, Teleport will be able to resolve
+the issues with adding AWS servers to Teleport clusters automatically.
+
+
+## Discovery
+
+A Teleport auth agent will need to be configured with an assume role policy for
+each of the accounts that the agent will discover services on as described by
+the EC2 join method docs[1]
+
+Discovery can use a matcher similar to the `db_service/aws` matcher:
+
+```yaml
+auth_service:
+  enabled: "yes"
+  aws:
```

Comment:
RDS discovery configuration is set up in database service. We should have SSH service or general proxy service that does this because of our cloud users. 

We want to minimize configuration changes for auth service - as cloud users can't edit those.

We would also need a corresponding `teleport ssh configure` command as well